packages/compass-editor/src/json-editor.tsx

```diff
@@ -1347,8 +1376,55 @@ async function setCodemirrorEditorValue(
   });
 }
 
+/**
+ * Retrieves the contents of an editor, use this with RTL like this:
+ *
+ * ```
+ * render(<Editor data-testid='my-editor' />);
+ * getCodemirrorEditorValue(screen.getByTestId('editor-test-id'));
+ * ```
+ */
+async function getCodemirrorEditorValue(
+  element: HTMLElement | string | null
+): Promise<string> {
+  if (typeof element === 'string') {
+    element = document.querySelector<HTMLElement>(`[data-testid="${element}"]`);
+  }
+  if (!element || !element.hasAttribute('data-codemirror')) {
+    throw new Error('Cannot find editor container');
+  }
+  const editorView = (element as HTMLElement & { _cm: EditorView })._cm;
+  await waitUntilEditorIsReady(editorView);
+
+  return editorView.state.sliceDoc() ?? '';
+}
+
+/**
+ * Clicks on an editor, use this with RTL like this:
+ *
+ * ```
+ * render(<Editor data-testid='my-editor' />);
+ * clickOnCodemirrorHandler(screen.getByTestId('editor-test-id'));
+ * ```
+ */
+async function clickOnCodemirrorHandler(
+  element: HTMLElement | string | null
+): Promise<void> {
+  if (typeof element === 'string') {
+    element = document.querySelector<HTMLElement>(`[data-testid="${element}"]`);
+  }
+  if (!element || !element.hasAttribute('data-codemirror')) {
+    throw new Error('Cannot find editor container');
+  }
+  const editorView = (element as HTMLElement & { _cm: EditorView })._cm;
+  editorView.focus();
+  await waitUntilEditorIsReady(editorView);
```

Comment:
To be honest it's a bit unexpected to me that we need to have this specific logic here, I think even in e2e tests we just click the editor to focus it and it seems to work fine, I don't mind having this if there is no other way around it, but fwiw it should be enough to `userEvent.click(editorSelector)` in the tests to make it work. If we need this one, let's clarify with a comment why we need it for future reference

Reply:
I see the comment added, will resolve